internal/storage/stats.go

```diff
@@ -120,79 +125,145 @@ func (stats *PrimaryKeyStats) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
-// updatePk update minPk and maxPk value
-func (stats *PrimaryKeyStats) updatePk(pk PrimaryKey) {
-	if stats.MinPk == nil {
-		stats.MinPk = pk
-	} else if stats.MinPk.GT(pk) {
-		stats.MinPk = pk
-	}
-
-	if stats.MaxPk == nil {
-		stats.MaxPk = pk
-	} else if stats.MaxPk.LT(pk) {
-		stats.MaxPk = pk
-	}
-}
-
-// StatsWriter writes stats to buffer
-type StatsWriter struct {
-	buffer []byte
-}
+func (stats *PrimaryKeyStats) UpdateByBinlogs(ctx context.Context, cm ChunkManager, binlogs []*datapb.Binlog) error {
```

Comment:
prefer not to reload binlog paths but simply merge multiple BF into a compound BF.

compound BF is simply multiple BFs and read applies on all BFs and if one of the BF exist the BF hit